### PR TITLE
Avoid triggering an assert when saving the park

### PR DESCRIPTION
**Pull request by ZehMatt**
The function RideAllocateAtIndex was previously called GetOrAllocateRideAt, it shouldn't use this function to get a ride when saving.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
It still crashes for me.

File: src/openrct2/park/ParkFile.cpp [hunk 1]
```diff
@@ -1193,7 +1193,12 @@ namespace OpenRCT2
                     // Ride ID
                     cs.ReadWrite(rideId);
 
-                    auto& ride = *RideAllocateAtIndex(rideId);
+                    auto& ride = [&]() -> Ride& {
+                        if (cs.GetMode() == OrcaStream::Mode::READING)
+                            return *GetRide(rideId);
+                        else
+                            return *RideAllocateAtIndex(rideId);
```
Comment (Gymnasiast) [inline src/openrct2/park/ParkFile.cpp:1200]: This looks like it’s the wrong way round - it uses `RideAllocateAtIndex` when _saving_.

Comment (ZehMatt) [reply src/openrct2/park/ParkFile.cpp:1200]: Oh yeah

**Review by Gymnasiast — APPROVED** [review]
This solves the problem for me.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**